src/ucp/wireup/wireup_ep.c

```diff
@@ -666,6 +667,20 @@ uct_ep_h ucp_wireup_ep_extract_next_ep(uct_ep_h uct_ep)
     return next_ep;
 }
 
+void ucp_wireup_ep_destroy_next_ep(ucp_wireup_ep_t *wireup_ep)
+{
+    uct_ep_h uct_ep;
+
+    ucs_assert(wireup_ep != NULL);
+    uct_ep = ucp_wireup_ep_extract_next_ep(&wireup_ep->super.super);
+    ucs_assert_always(uct_ep != NULL);
+    uct_ep_destroy(uct_ep);
+
+    wireup_ep->flags &= ~UCP_WIREUP_EP_FLAG_LOCAL_CONNECTED;
+    ucs_assert(wireup_ep->flags == 0);
+    ucp_proxy_ep_set_uct_ep(&wireup_ep->super, NULL, 0);
```

Comment:
I guess it should be done from `ucp_wireup_ep_extract_next_ep`

Reply:
not sure it's always needed there, since it wasn't needed there till now (and maybe not even now).